build/configs/artik05x/artik05x_download.sh

```diff
@@ -167,28 +167,30 @@ download()
 
 erase()
 {
+	# Generate Partition Map
+	${SCRIPTS_PATH}/partition_gen.sh
+
+	pushd ${OPENOCD_DIR_PATH} > /dev/null
+
 	case $optarg in
 		OTA|ota)
 			PART_NAME=ota
+			${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c "flash_erase_part ${PART_NAME}; exit" || exit 1
```

Comment:
Repeatation looks not good.
For example, how about below?
```
  case ..
    OTA|ota)
      PART_NAME=ota
      ;;
    USERFS|userfs)
      PART_NAME=user
      ;;
    ALL|all)
      PART_NAME=all
      NEED_PROTECTCMD=y
      ;;
    *)
      ....
  esac

  if [ "${NEED_PROTECTCMD}" == "y" ]; then
    PROTECT_OFFCMD="flash_protect off"
    PROTECT_ONCMD="flash_protect on"
  fi

 # Generate ...
 ${SCRIPTS_PATH}/....

  # Download all ...
  pushd ...
  ${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c "${PROTECT_OFFCMD}; flash_erase_part ${PART_NAME}; ${PROTECT_ONCMD}; exit" || exit 1
  popd ...
```

Reply:
This gives the flexibility to modify PATH, CMDs and attributes(present and future) to each partition.
But if reviewer wants to tie and bound all attributes(present and future) , erase functionalities, etc to one command then this comment can be incorporated.  (As a matter of fact, its all about flexibility, readability and maintainability vs repetitions). Anything is okay.